In [1]:
import snap

LoadedGraph = snap.LoadEdgeList(snap.PNEANet, "/data/06271/cju256/money_flow_weighted_edgelist.csv", 0, 1, ',')

GraphNodes = len(list(LoadedGraph.Nodes()))
GraphVertices = len(list(LoadedGraph.Edges()))

print(GraphNodes)
print(GraphVertices)

23097596
131705390


In [2]:
#GraphClustCoeff = snap.GetClustCf(LoadedGraph, -1)
#print(GraphClustCoeff)

Clustering Coeffient on Unique Edges only. Edges are flow of money. No null nodes:
CC: 0.11253357075275515
Nodes: 23097596
Edges: 131705390

Clustering Coeffient on all Edges. Edges are flow of money. No null nodes:
CC: 0.017942184397686606
Nodes: 23097596
Edges: 341309788

In [4]:
Components_s = snap.TCnComV()
snap.GetSccs(LoadedGraph, Components_s)
print("Number of SC Components: ", len(Components_s))

Number of SC Components:  9557171


In [5]:
MxScc = snap.GetMxScc(LoadedGraph)
print("max scc nodes %d, edges %d" % (MxScc.GetNodes(), MxScc.GetEdges()))

max scc nodes 13257479, edges 115691467


In [8]:
13257479/23097596

0.5739765731464002

In [6]:
Components = snap.TCnComV()
snap.GetWccs(LoadedGraph, Components)
print("Number of WC Components: ", len(Components))

Number of WC Components:  444543


In [7]:
MxWcc = snap.GetMxWcc(LoadedGraph)
print("max wcc nodes %d, edges %d" % (MxWcc.GetNodes(), MxWcc.GetEdges()))

WccV = snap.TIntPrV()

snap.GetWccSzCnt(LoadedGraph, WccV)

print("# of connected component sizes", WccV.Len())
for comp in WccV:
    print("size %d, number of components %d" % (comp.GetVal1(), comp.GetVal2()))

max wcc nodes 22067849, edges 131055776
# of connected component sizes 39
size 1, number of components 2
size 2, number of components 357103
size 3, number of components 59352
size 4, number of components 16693
size 5, number of components 5935
size 6, number of components 2503
size 7, number of components 1278
size 8, number of components 647
size 9, number of components 356
size 10, number of components 229
size 11, number of components 128
size 12, number of components 77
size 13, number of components 60
size 14, number of components 50
size 15, number of components 29
size 16, number of components 24
size 17, number of components 10
size 18, number of components 10
size 19, number of components 11
size 20, number of components 9
size 21, number of components 5
size 22, number of components 2
size 23, number of components 5
size 24, number of components 3
size 25, number of components 1
size 26, number of components 2
size 27, number of components 2
size 28, number of components 1
s

In [10]:
22067849/23097596

0.9554175681313328